app/scripts/ui.js

```diff
@@ -24,34 +24,37 @@ import { setupMultiplex } from './lib/stream-utils';
 import { getEnvironmentType } from './lib/util';
 import metaRPCClientFactory from './lib/metaRPCClientFactory';
 
-start().catch(log.error);
-
-async function start() {
-  async function displayCriticalError(container, err, metamaskState) {
-    const html = await getErrorHtml(SUPPORT_LINK, metamaskState);
-
-    container.innerHTML = html;
+const container = document.getElementById('app-content');
 
-    const button = document.getElementById('critical-error-button');
+const WORKER_KEEP_ALIVE_INTERVAL = 1000;
```

Comment:
If the worker is killed after 5 minutes of inactivity, do we need to ping it every second? What's the reason behind this?

Reply:
I want it to receive this ping as soon as it goes inactive. 5 minutes is the approximate time after which service worker dies but it is not guaranteed interval.

Reply:
My thought was that sending the `WORKER_KEEP_ALIVE_MESSAGE` message resets the 5 minute-ish timer. If this is the case then I think it can be safe to move it up to 10 seconds, 30 seconds, or even a minute. Is this not the case?

Reply:
Are you sure it dies after 5 minutes even if it's receiving messages during that period of time? If not, even a message every 4 minutes makes it immortal.

Reply:
Jyoti says it's not guaranteed, so 4 minutes seems like we'd be living on the edge

Reply:
I've seen situations where 1s was helpful. Can confirm.